src/EditorFeatures/CSharpTest2/Recommendations/WhereKeywordRecommenderTests.cs

```diff
@@ -297,6 +297,27 @@ public async Task TestNoWhereAfterDot()
             await VerifyAbsenceAsync(
 @"public class Goo<where> : System.$$
 {
+}");
+        }
+
+        [WorkItem(30785, "https://github.com/dotnet/roslyn/issues/30785")]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterDot1()
+        {
+            await VerifyAbsenceAsync(
+@"public class C
+{
+    void M<T> where T : System.$$
+}");
+        }
+
+        [WorkItem(30785, "https://github.com/dotnet/roslyn/issues/30785")]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterDot2()
+        {
+            await VerifyAbsenceAsync(
+@"public class C<T> where T : System.$$
```

Comment:
consider test for `where T : global::` and `where T : List<$$`,  `where T : List<(int, $$` etc.